.github/workflows/lint-and-unit-test.yml

```diff
@@ -0,0 +1,41 @@
+name: Lint and unit test
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '**/**.go'
+  pull_request:
+    paths:
+      - '**/**.go'
+
+jobs:
+  lint-and-unit-test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.21
+    
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo add-apt-repository universe && sudo apt-get install -y libswscale-dev libavcodec-dev libavformat-dev libavutil-dev ffmpeg libfuse2
```

Comment:
Do we still need `libswscale-dev libavcodec-dev libavformat-dev libavutil-dev ffmpeg`?

Reply:
I _think_ what's happening here is that the linter compiles cgo on each run. I spent a good amount of time finding workarounds to having these installed via apt, but ended up giving up since it's just for lint and tests in a temp runner environment

![Screenshot 2024-04-18 at 2 58 10 PM](https://github.com/erh/viamrtsp/assets/55464069/b6fbbe3b-9bff-4903-94bd-c879b983ca61)

ffmpeg is not needed though I was able to remove that, but `libswscale-dev libavcodec-dev libavformat-dev libavutil-dev` are necessary

Reply:
We should be linking against the version of ffmpeg we use in the static build even in tests. Otherwise, we are testing something that might have nothing to do with how our application actually behaves.

I think this can be fixed in the following way:

1. Export `PKG_CONFIG_PATH` and `CGO_LDFLAGS` as env vars in the makefile so they are available to all targets
2. Remove hard coded `PKC_CONFIG_PATH` and `CGO_LDFLAGS` env vars from targets as they are now global to all targets
3. Add `test` to the `.PHONY` so tests actually run

```
diff --cc Makefile
index d1cb587,300c2cd..0000000
--- a/Makefile
+++ b/Makefile
@@@ -23,13 -23,13 +23,13 @@@ CGO_LDFLAGS := -L$(FFMPEG_BUILD)/li
  ifeq ($(UNAME_S),Linux)
  	CGO_LDFLAGS := "$(CGO_LDFLAGS) -l:libjpeg.a"
  endif
 -
+ export PKG_CONFIG_PATH=$(FFMPEG_BUILD)/lib/pkgconfig
+ export CGO_LDFLAGS
 -.PHONY: build-ffmpeg lint test update-rdk module clean clean-all
 +
- .PHONY: build-ffmpeg tool-install gofmt lint update-rdk module clean clean-all
++.PHONY: build-ffmpeg tool-install gofmt test lint update-rdk module clean clean-all
  
  $(BIN_OUTPUT_PATH)/viamrtsp: build-ffmpeg *.go cmd/module/*.go
- 	PKG_CONFIG_PATH=$(FFMPEG_BUILD)/lib/pkgconfig \
- 		CGO_LDFLAGS=$(CGO_LDFLAGS) \
- 		go build -o $(BIN_OUTPUT_PATH)/viamrtsp cmd/module/cmd.go
+ 	go build -o $(BIN_OUTPUT_PATH)/viamrtsp cmd/module/cmd.go
  
  tool-install:
  	GOBIN=`pwd`/$(TOOL_BIN) go install \
```

After that you should no longer need to install apt packages to run lint